src/gp_activetable.c

```diff
@@ -1030,35 +995,21 @@ load_table_size(HTAB *local_table_stats_map)
  * This function is used to prepare the second array parameter
  * of function diskquota_fetch_table_stat.
  */
-static StringInfoData
-convert_map_to_string(HTAB *local_active_table_oid_maps)
+static void
+convert_map_to_string(HTAB *local_active_table_oid_maps, StringInfoData *active_oids)
```

Comment:
Maybe don't change this function since we do not use it anyway?

Reply:
It is used below in `pull_active_list_from_seg`, but I can remove it and put its code into `pull_active_list_from_seg`.

Reply:
Yes, probably it would me orptimal to get rid of local_active_table_oid_map as well

Reply:
> it would be quite optimal to get rid of local_active_table_oid_map as well

But this, unfortunately, will not work, because we need unique oids from different segments, for the collection of which the `local_active_table_oid_map` hashmap is used.

Reply:
Then why do we change the `pull_active_list_from_seg` and remove `convert_map_to_string`? Maybe make `gp_fetch_active_tables` function return stringinfo, which is initialized inside? Like
```
StringInfoData
gp_fetch_active_tables(bool is_init)
{
	HTAB          *local_active_table_oid_maps;
	StringInfoData active_oid_list;

	Assert(Gp_role == GP_ROLE_DISPATCH);

	if (is_init)
	{
		initStringInfo(&active_oid_list);
		load_table_size(&active_oid_list);
	}
	else
	{
		/* step 1: fetch active oids from all the segments */
		local_active_table_oid_maps = pull_active_list_from_seg();
		active_oid_list             = convert_map_to_string(local_active_table_oid_maps);

		ereport(DEBUG1,
		        (errcode(ERRCODE_INTERNAL_ERROR), errmsg("[diskquota] active_old_list = %s", active_oid_list.data)));

		/* step 2: fetch active table sizes based on active oids */
		pull_active_table_size_from_seg(active_oid_list.data);

		hash_destroy(local_active_table_oid_maps);
	}
	return active_oid_list;
}

```

Reply:
> why do we change the `pull_active_list_from_seg` and remove `convert_map_to_string`? Maybe make `gp_fetch_active_tables` function return stringinfodata, which is initialized inside? Like

Because now I allocate and free data for `StringInfoData` in the same function.

Reply:
> I allocate and free data for StringInfoData in the same function

And this is the best and the rightest  pattern for using StringIngoData?

Reply:
Then maybe leave convert_map_to_string as it was? and append its string info to the original one? 
For now it feels really dirty to carry the stringinfo param from function to function.

Reply:
> And this is the best and the rightest pattern for using StringInfoData?

yes

Reply:
> and append its string info to the original one

This will create a lot of unnecessary allocations!